src/hash.c

```diff
@@ -80,6 +80,8 @@ void ntru_sha256_8way_nosimd(uint8_t *input[8], uint16_t input_len, uint8_t *dig
 
 void ntru_set_optimized_impl_hash() {
 #ifdef NTRU_DETECT_SIMD
+#ifdef __clang__
```

Comment:
Are you disabling all SSE/AVX2 support on Mac OS? If so, why not do it through the SIMD switch in Makefile.osx?
Is there a reason why you're using an empty #ifdef and an #else instead of an #ifndef?